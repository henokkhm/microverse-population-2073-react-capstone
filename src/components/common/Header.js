import { Link, useLocation } from 'react-router-dom';

import styles from '../../styles/Header.module.css';
import hamburger from '../../assets/icons/hamburger.svg';
import back from '../../assets/icons/back.svg';
import mic from '../../assets/icons/mic.svg';
import settings from '../../assets/icons/settings.svg';

function Header() {
  const location = useLocation();

  const backButton = (
    <Link to="/" data-testid="back-button">
      <button className={styles.iconBtn} type="button">
        <img src={back} alt="go back button" />
      </button>
    </Link>
  );

  const hamburgerButton = (
    <button className={styles.iconBtn} type="button">
      <img src={hamburger} alt="hamburger menu icon" />
    </button>
  );

  return (
    <header className={styles.appHeader}>
      {location.pathname === '/' ? hamburgerButton : backButton}
      <h1 className={styles.pageTitle}>{location.pathname === '/' ? 'Regions' : 'Countries'}</h1>
      <button className={styles.iconBtn} type="button">
        <img src={mic} alt="microphone icon" />
      </button>
      <button className={styles.iconBtn} type="button">
        <img src={settings} alt="settings menu icon" />
      </button>
    </header>
  );
}

export default Header;
