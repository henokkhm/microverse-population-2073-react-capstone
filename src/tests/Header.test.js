import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Header from '../components/common/Header';

describe('The Header component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
