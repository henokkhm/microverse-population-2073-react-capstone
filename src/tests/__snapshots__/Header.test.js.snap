// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="appHeader"
  >
    <button
      class="iconBtn"
      type="button"
    >
      <img
        alt="hamburger menu icon"
        src="hamburger.svg"
      />
    </button>
    <h1
      class="pageTitle"
    >
      Regions
    </h1>
    <button
      class="iconBtn"
      type="button"
    >
      <img
        alt="microphone icon"
        src="mic.svg"
      />
    </button>
    <button
      class="iconBtn"
      type="button"
    >
      <img
        alt="settings menu icon"
        src="settings.svg"
      />
    </button>
  </header>
</DocumentFragment>
`;
